### DISCUSSION TITLE
How can I embed BDFR in my script?

### DISCUSSION THREAD

**Discussion by cenk78** [Serene-Arc/bulk-downloader-for-reddit#329] [Q&A]
As the title says, how can I use BDFR inside my script? I am kind of new to python. Couldn't find a way.

**Comment by Serene-Arc** [comment]
What kind of script do you mean? With the BDFR in another Python script, or something like PowerShell/Bash?

**Reply by cenk78** [reply]
In another python script. I'm going to download reddit posts and upload them to instagram. Need BDFR in a py script.

**Reply by Serene-Arc** [reply]
Ah, okay, well we don't have an official API written so far so you'll have to check out the source code, but you can use the classes provided to parse the Reddit submissions. You can write custom values to the `Configuration` class and then use that and the `RedditDownloader` class to get a collection of `Resource` objects with which you can do whatever you want.

**Reply by cenk78** [reply]
Ok. Thank you.

**Comment by aliparlakci** [comment]
At the moment, we don't have any public API but we probably will sooner or later. Until then, you can use bdfr as a separate process in your python program.
